src/makielayout/blocks/colorbar.jl

```diff
@@ -27,11 +27,28 @@ end
 
 function Colorbar(fig_or_scene, plot::AbstractPlot; kwargs...)
     colorbar_check((:colormap, :limits, :highclip, :lowclip), keys(kwargs))
-
     if haskey(plot, :calculated_colors) && plot.calculated_colors[] isa ColorMap
         cmap = plot.calculated_colors[]
         scale = cmap.scale
     else
+        # Atomic plots should always have calculated_colors set, if they actually use a colormap
+        func = plotfunc(plot)
+        if func in Makie.atomic_functions
+            error("Plot `$(func)` doesn't have a color attribute that uses a colormap to map to colors, so it cannot be used to create a colorbar. Found color attribute: $(typeof(plot.calculated_colors[]))). Needs to be a number of vector of numbers!")
+        end
+        # So this is for recipes
+        if !all(x-> haskey(plot, x), (:colormap, :colorrange, :highclip, :lowclip))
+            attributes = filter(x-> haskey(plot, x), [:colormap, :colorrange, :highclip, :lowclip])
+            error("Plot $(func) doesn't have all colormap attributes. Attributes needed: colormap, colorrange, highclip, lowclip. Found: $(attributes)")
+        end
+        if to_value(plot.colorrange) isa Automatic || (to_value(get(plot, :color, nothing)) isa Union{AbstractVector{<: Colorant}, Colorant})
+            error("""
+            Plot $(func) doesn't have a color attribute that uses a colormap to map to colors, so it cannot be used to create a colorbar.
+            Please create the colorbar manually e.g. via `Colorbar(f[1, 2], colorrange=the_range, colormap=the_colormap)`.
+            This could also mean, that you're using a recipe, which itself doesn't calculate the colorrange, and leaves it to its children.
```

Comment:
Did this not work before (the barplot example)? Would seem to be a big regression if so.

Reply:
It was indeed able to create one, but it created the _wrong_ Colorbar before, since it didn't connect the colorrange, which only happens for the meshplot inside.

Reply:
hm one way or another I think it should work, it's very weird having to access the nested plots, no? maybe an internal function specifying the child plot with the color that gets called by the `Colorbar` constructor?

Reply:
I mean we could search for subplots that do have a complete colormap... Now with `calculated_colors`, it should be pretty clear, that such a plot _does_ have numbers that are mapped to colors... The question is, how do we deal with multiple subplots? 
We could check if they use the same `calculated_colors` and if not error...and if it's just one it should quite clearly be the plot to be used for Colorbar

Reply:
Maybe a combination, like with data_limits? Some plots need to overload `data_limits` but there is a fallback for all recipes, that usually does the correct thing by calculating `data_limits` of the children